diff --color -ruN ./og/cmake/config-ix.cmake ./mod/cmake/config-ix.cmake
--- ./og/cmake/config-ix.cmake	2025-12-20 20:57:21.863799894 +0600
+++ ./mod/cmake/config-ix.cmake	2025-12-20 21:01:23.985676097 +0600
@@ -173,7 +173,7 @@
     # library on a 64-bit system which would result in a link-time failure.
     cmake_push_check_state()
     list(APPEND CMAKE_REQUIRED_INCLUDES ${ZLIB_INCLUDE_DIRS})
-    list(APPEND CMAKE_REQUIRED_LIBRARIES ${ZLIB_LIBRARY})
+    list(APPEND CMAKE_REQUIRED_LIBRARIES ${ZLIB_LIBRARIES})
     check_symbol_exists(compress2 zlib.h HAVE_ZLIB)
     cmake_pop_check_state()
     if(LLVM_ENABLE_ZLIB STREQUAL FORCE_ON AND NOT HAVE_ZLIB)
@@ -279,11 +279,11 @@
     # Skip libedit if using ASan as it contains memory leaks.
     if (LLVM_ENABLE_LIBEDIT AND NOT LLVM_USE_SANITIZER MATCHES ".*Address.*")
       if(LLVM_ENABLE_LIBEDIT STREQUAL FORCE_ON)
-        find_package(LibEdit REQUIRED)
+        find_package(editline REQUIRED)
       else()
-        find_package(LibEdit)
+        find_package(editline)
       endif()
-      set(HAVE_LIBEDIT "${LibEdit_FOUND}")
+      set(HAVE_LIBEDIT "${editline_FOUND}")
     else()
       set(HAVE_LIBEDIT 0)
     endif()
diff --color -ruN ./og/lib/LineEditor/CMakeLists.txt ./mod/lib/LineEditor/CMakeLists.txt
--- ./og/lib/LineEditor/CMakeLists.txt	2025-12-20 20:57:12.287583069 +0600
+++ ./mod/lib/LineEditor/CMakeLists.txt	2025-12-20 21:01:43.539366371 +0600
@@ -1,5 +1,5 @@
 if(HAVE_LIBEDIT)
-  set(link_libs LibEdit::LibEdit)
+  set(link_libs editline::editline)
 endif()
 
 add_llvm_component_library(LLVMLineEditor
diff --color -ruN ./og/lib/Support/CMakeLists.txt ./mod/lib/Support/CMakeLists.txt
--- ./og/lib/Support/CMakeLists.txt	2025-12-20 20:57:37.766144259 +0600
+++ ./mod/lib/Support/CMakeLists.txt	2025-12-20 21:02:59.738720910 +0600
@@ -330,27 +330,13 @@
 # This block is only needed for llvm-config. When we deprecate llvm-config and
 # move to using CMake export, this block can be removed.
 if(LLVM_ENABLE_ZLIB)
-  # CMAKE_BUILD_TYPE is only meaningful to single-configuration generators.
-  if(CMAKE_BUILD_TYPE)
-    string(TOUPPER ${CMAKE_BUILD_TYPE} build_type)
-    get_property(zlib_library TARGET ZLIB::ZLIB PROPERTY LOCATION_${build_type})
-  endif()
-  if(NOT zlib_library)
-    get_property(zlib_library TARGET ZLIB::ZLIB PROPERTY LOCATION)
-  endif()
+  set(zlib_library ${ZLIB_LIBRARIES})
   get_library_name(${zlib_library} zlib_library)
   set(llvm_system_libs ${llvm_system_libs} "${zlib_library}")
 endif()
 
 if(LLVM_ENABLE_ZSTD)
-  # CMAKE_BUILD_TYPE is only meaningful to single-configuration generators.
-  if(CMAKE_BUILD_TYPE)
-    string(TOUPPER ${CMAKE_BUILD_TYPE} build_type)
-    get_property(zstd_library TARGET ${zstd_target} PROPERTY LOCATION_${build_type})
-  endif()
-  if(NOT zstd_library)
-    get_property(zstd_library TARGET ${zstd_target} PROPERTY LOCATION)
-  endif()
+  set(zstd_library ${zstd_LIBRARIES})
   if (zstd_target STREQUAL zstd::libzstd_shared)
     get_library_name(${zstd_library} zstd_library)
     set(llvm_system_libs ${llvm_system_libs} "${zstd_library}")
diff --color -ruN ./og/lib/WindowsManifest/CMakeLists.txt ./mod/lib/WindowsManifest/CMakeLists.txt
--- ./og/lib/WindowsManifest/CMakeLists.txt	2025-12-20 20:57:56.577527061 +0600
+++ ./mod/lib/WindowsManifest/CMakeLists.txt	2025-12-20 21:03:45.681819618 +0600
@@ -21,14 +21,7 @@
 # This block is only needed for llvm-config. When we deprecate llvm-config and
 # move to using CMake export, this block can be removed.
 if(LLVM_ENABLE_LIBXML2)
-  # CMAKE_BUILD_TYPE is only meaningful to single-configuration generators.
-  if(CMAKE_BUILD_TYPE)
-    string(TOUPPER ${CMAKE_BUILD_TYPE} build_type)
-    get_property(libxml2_library TARGET LibXml2::LibXml2 PROPERTY LOCATION_${build_type})
-  endif()
-  if(NOT libxml2_library)
-    get_property(libxml2_library TARGET LibXml2::LibXml2 PROPERTY LOCATION)
-  endif()
+  set(libxml2_library ${libxml2_LIBRARIES})
   get_library_name(${libxml2_library} libxml2_library)
   set_property(TARGET LLVMWindowsManifest PROPERTY LLVM_SYSTEM_LIBS ${libxml2_library})
 endif()
